### PR TITLE
homework2

### PR DESCRIPTION
**Pull request by dkh**

### CODE REVIEW

**Review by oleksandr-kychun — COMMENTED** [review]
Nice solution) Object model looks correct.
About state_and_behaviour.rb:
When you work with hash, it is preferable to use hsh.fetch method. It can help with defaults.
Car.default_car will raise error but it shouldn't.
Besides, it is preferable to use getters/setters instead of dealing with instance variables.

File: lib/topics/state_and_behaviour.rb [hunk 1]
```diff
@@ -2,4 +2,27 @@
...
 module StateAndBehaviour
+  # StateAndBehaviour::Car
+  class Car
+    attr_accessor :year, :color, :model, :current_speed
+
+    def self.default_car
+      new {}
+    end
+
+    def initialize(attributes)
```
Comment (sveredyuk) [inline lib/topics/state_and_behaviour.rb]: attributes = {}

File: lib/topics/state_and_behaviour.rb [hunk 2]
```diff
@@ -2,4 +2,27 @@
...
+      self.year          = attributes.fetch(:year, 2000)
```
Comment (sveredyuk) [inline lib/topics/state_and_behaviour.rb]: use @year instead accessor (accessors are bad)

File: lib/topics/state_and_behaviour.rb [hunk 3]
```diff
@@ -2,4 +2,27 @@
...
+      self.model         = attributes.fetch(:model, 'skyline')
+      self.current_speed = 0
+    end
+
+    def speed_up(speed)
+      self.current_speed += speed
+    end
+
+    def push_break(speed)
+      self.current_speed -= speed unless self.current_speed <= speed
```
Comment (sveredyuk) [inline lib/topics/state_and_behaviour.rb]: hard readable line try ti return from method unless this condition
and then only execute what you need

**Pull request closed without merging**